charmhelpers/contrib/openstack/reactive.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Canonical
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library of base classes for common reactive endpoints and relations."""
+
+import json
+
+import charms.reactive as reactive
+
+import charmhelpers.core.hookenv as hookenv
+
+
+class OpenStackRelationBase(reactive.RelationBase):
```

Comment:
Why RelationBase rather than Endpoint?

Reply:
The change is to be used with the [keystone-domain-backend](https://github.com/openstack/charm-interface-keystone-domain-backend) and [cinder-backend](https://github.com/openstack/charm-interface-cinder-backend) interfaces which both are RelationBase interfaces. As soon as we need to use this with Endpoint-based interfaces we should add a separate definition for that.